locale.c

```diff
@@ -664,7 +664,8 @@ S_positional_newlocale(int mask, const char * locale, locale_t base)
  * in C.  This macro substitutes C for the locale appropriately, expanding to
  * nothing on the more typical case where all possible categories present on
  * the platform are handled. */
-#  ifdef HAS_IGNORED_LOCALE_CATEGORIES_
+#  if defined(HAS_IGNORED_LOCALE_CATEGORIES_)       \
```

Comment:
Point of information for a non-C programmer:  Is there a difference between this syntax:
```
# ifdef SOMETHING
```
and this syntax:
```
# if defined(SOMETHING)
```
Why do you use one or the other?

Reply:
`#ifdef` can only deal with one singe symbol
`#if defined(FOO)` is equal to `#ifdef FOO`, but when testing on more than one single symbol, like in this code, the second syntax is required
`#if defined(FOO) || defined(BAR)` does not have an `#ifdef` variant

Reply:
`#ifdef FOO` is subsumed by the form `#if defined FOO` (or equivalently, `#if defined ( FOO )`). As for why both forms exist, I believe `#ifdef` (and `#ifndef`) came first historically and `defined` was added later when people needed more complex conditionals.

Choosing between them is mostly a matter of taste (personally I don't like the extra parentheses in the `defined (FOO)` form, for example). Some purists argue that you should limit yourself to `#ifdef` because anything more complex than that is a sign that you're abusing the preprocessor for things it was never meant to do (and you should presumably move complex configuration logic to a separate stage that emits simple macro definitions). In perl, that train has sailed.